### PR TITLE
Add functional tests

### PR DESCRIPTION
**Pull request by emillon**
This loads a (fake) PKCS11 DLL and calls some functions on it.

Marking this as WIP because this is still fairly minimal.

### CODE REVIEW

File: test/test_functional.ml [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+  let (module R) = driver in
+  R.initialize ();
+  [ "Driver" >::: test_driver driver;
+    "Session" >::: test_session driver;
+    "Object" >::: test_object driver;
+  ]
+
+let suite =
+  "Functional" >::: [
+    "Load P11" >::: test_p11_load;
```
Comment (bbc2) [inline test/test_functional.ml]: There's a mix of comma-first and other styles.  Full comma-first style would be preferable no?

**Review by bbc2 — APPROVED** [review]
Nice!

File: test/test_functional.ml (hunk 1 above)
Comment (emillon) [reply test/test_functional.ml]: Indeed - just fixed this.

**Comment by emillon** [comment]
@bbc2 fixed - let me know what you think.

**Pull request merged**